### ISSUE TITLE
[BUG] encoder head's size mismatch error

### ISSUE THREAD

**Issue by sascholle** [clessig/atmorep#90]
**Describe the bug**
Size mismatch error when running the multi3-uv model: model_id3cizyl1q (see screenshot below)

_0:     size mismatch for encoders.0.heads.1.proj_out.weight: copying a param with shape torch.Size([2048, 2560]) from checkpoint, the shape in current model is torch.Size([2048, 2816])._ 

The same error for encoder 1, heads 0 through 9 which correspond to the two wind fields, and then a similar but different set of mismatched dimensions for the temperature field: 

_1: 	size mismatch for encoders.2.heads.0.proj_out.weight: copying a param with shape torch.Size([1024, 1280]) from checkpoint, the shape in current model is torch.Size([1024, 1408])._

Again for heads 0 through 9.  

The input dim is correct, but the second dim seems to be scaled by 1.25, and I can't find the parameter to inform this.  The obvious fixes of changing the encoder num heads or token size/number doesn't seem to work. 

**To Reproduce**
Running the 3-field model with the downloaded model config: 

{
    "num_accs_per_task": 4,
    "with_hvd": true,
    "hvd_rank": 0,
    "hvd_size": 16,
    "back_passes_per_step": 4,
    "comment": "",
    "file_format": "grib",
    "data_dir": "../data/",
    "level_type": "ml",
    "fields": [
        [
            "velocity_u",
            [
                1,
                2048,
                [
                    "velocity_v",
                    "temperature"
                ],
                0,
                [
                    "3k6e6p7o",
                    141
                ]
            ],
            [
                96,
                105,
                114,
                123,
                137
            ],
            [
                12,
                6,
                12
            ],
            [
                3,
                9,
                9
            ],
            [
                0.5,
                0.9,
                0.2,
                0.05
            ],
            "local"
        ],
        [
            "velocity_v",
            [
                1,
                2048,
                [
                    "velocity_u",
                    "temperature"
                ],
                1,
                [
                    "brxmevmt",
                    141
                ]
            ],
            [
                96,
                105,
                114,
                123,
                137
            ],
            [
                12,
                6,
                12
            ],
            [
                3,
                9,
                9
            ],
            [
                0.5,
                0.9,
                0.2,
                0.05
            ],
            "local"
        ],
        [
            "temperature",
            [
                1,
                1024,
                [
                    "velocity_u",
                    "velocity_v"
                ],
                2,
                [
                    "1za45vud",
                    155
                ]
            ],
            [
                96,
                105,
                114,
                123,
                137
            ],
            [
                12,
                6,
                12
            ],
            [
                3,
                9,
                9
            ],
            [
                0.5,
                0.9,
                0.2,
                0.05
            ],
            "local"
        ]
    ],
    "fields_prediction": [
        [
            "velocity_u",
            0.4
        ],
        [
            "velocity_v",
            0.4
        ],
        [
            "temperature",
            0.2
        ]
    ],
    "fields_targets": [],
    "years_train": [
        1980,
        1981,
        1982,
        1983,
        1984,
        1985,
        1986,
        1987,
        1988,
        1989,
        1990,
        1991,
        1992,
        1993,
        1994,
        1995,
        1996,
        1997,
        1998,
        1999,
        2000,
        2001,
        2002,
        2003,
        2004,
        2005,
        2006,
        2007,
        2008,
        2009,
        2010,
        2011,
        2012,
        2013,
        2014,
        2015,
        2016,
        2017
    ],
    "years_test": [
        2018
    ],
    "month": null,
    "geo_range_sampling": [
        [
            -90.0,
            90.0
        ],
        [
            0.0,
            360.0
        ]
    ],
    "time_sampling": 1,
    "data_smoothing": 0,
    "file_shape": [
        -1,
        721,
        1440
    ],
    "num_t_samples": 744,
    "num_patches_per_t_train": 4,
    "num_patches_per_t_test": 1,
    "torch_seed": 2098940104253649770,
    "batch_size_test": 8,
    "batch_size_start": 3,
    "batch_size_max": 3,
    "batch_size_delta": 8,
    "num_epochs": 5,
    "num_files_train": 1,
    "num_files_test": 1,
    "loader_num_workers": 2,
    "size_token_info": 8,
    "size_token_info_net": 16,
    "grad_checkpointing": true,
    "with_cls": false,
    "with_layernorm": true,
    "coupling_num_heads_per_field": 3,
    "dropout_rate": 0.05,
    "learnable_mask": false,
    "with_qk_lnorm": true,
    "num_neighborhoods": 2,
    "encoder_num_layers": 10,
    "encoder_num_heads": 16,
    "encoder_num_mlp_layers": 2,
    "encoder_att_type": "dense",
    "decoder_num_layers": 10,
    "decoder_num_heads": 16,
    "decoder_num_mlp_layers": 2,
    "decoder_self_att": false,
    "decoder_cross_att_ratio": 0.5,
    "decoder_cross_att_rate": 1.0,
    "decoder_att_type": "dense",
    "net_tail_num_nets": 16,
    "net_tail_num_layers": 0,
    "ensemble_weighted": false,
    "ensemble_weighted_T": 50.0,
    "losses": [
        "mse_ensemble",
        "stats"
    ],
    "lr_start": 1e-05,
    "lr_max": 2e-05,
    "lr_min": 1e-05,
    "weight_decay": 0.025,
    "lr_decay_rate": 1.025,
    "lr_start_epochs": 3,
    "lat_sampling_weighted": true,
    "BERT_strategy": "BERT",
    "BERT_window": false,
    "BERT_fields_synced": false,
    "BERT_mr_max": 2,
    "log_test_num_ranks": 0,
    "save_grads": false,
    "profile": false,
    "test_initial": true,
    "rng_seed": null,
    "with_wandb": true,
    "slurm_job_id": "7916478",
    "wandb_id": "3cizyl1q"
}

**Screenshots**

![Image](https://github.com/user-attachments/assets/be463f90-c23a-4fc5-8f77-55c78d317716)

**Hardware and environment:**
 - System: Levante
 - Runtime: sbatch
 - Python Version: 3.10.10
